Soluţii pentru Python Labs

astronos2007 opened the pull request:
Acest pull request conţine munca mea asiduă.

Probleme rezolvate:
- [x] Caesar
- [x] ICAO
- [x] Unic
- [x] Cursor
- [x] Fill shape
- [x] Paranteze
- [x] Tree
- [x] Syncronize

Commit:
Acesta este începutul sfârşitului pentru mine.

Commit:
Commit initial

alexcoman commented on python/solutii/cosmin_crihan/README.md:
```diff
@@ -0,0 +1 @@
+În acest director se regăseşte munca mea asiduă din ultimele 30 de secunde.
```
Cred că ar trebui să depui mai mult de 30 de secunde pentru acest laborator.

astronos2007 replied:
@alexandrucoman , ai putea să te uiţi te rog când ai timp peste codul meu? Am terminat problemele propuse şi aş dori nişte feedback :)

Commit:
Am rezolvat problemele "unic", "caesar" si "paranteze".

Commit:
Am rezolvat problema "icao".

Commit:
Am rezolvat problema "paint".

Commit:
Am rezolvat problema "tree".

Commit:
Am rezolvat problema cu sincronizarea uni si bidirectionala intre 2 foldere.

alexcoman commented on python/solutii/cosmin_crihan/caesar/caesar.py:
```diff
@@ -0,0 +1,98 @@
+un mesaj prin înlocuirea fiecărei litere cu litera de pe poziția aflată
+la n pași de ea în alfabet (unde este n este un număr întreg cunoscut)
+"""
+# pylint: disable=unused-argument
```
Această linie nu mai este necesară.

alexcoman commented on python/solutii/cosmin_crihan/caesar/caesar.py:
```diff
@@ -0,0 +1,98 @@
+            else:
+                index_litera = (ord(litera) - ord('a') + n) % 26
+                litera_decriptata = chr(index_litera + ord('a'))
+            decriptat += litera_decriptata
```
Nu este neapărat o practică bună. Mai multe detalii poți găsi mai jos:
- https://wiki.python.org/moin/PythonSpeed/PerformanceTips#String_Concatenation
- http://leadsift.com/python-string-concatenation/
- https://waymoot.org/home/python_string/

alexcoman commented on python/solutii/cosmin_crihan/caesar/caesar.py:
```diff
@@ -0,0 +1,98 @@
+
+        print("%s\n" % decriptat)
+
+        plauzibil = input('Plauzibil? (da=1/nu=0) ')  # intrebam utilizatorul daca mesajul e lizibil
```
Puteți găsi o altă soluție pentru a determina cheia.

alexcoman commented on python/solutii/cosmin_crihan/caesar/caesar.py:
```diff
@@ -0,0 +1,98 @@
+    return decriptat
+
+def main():
+
```
Linia aceasta nu este necesară.

alexcoman commented on python/solutii/cosmin_crihan/caesar/caesar.py:
```diff
@@ -0,0 +1,98 @@
+    try:
+        mesaje_decriptate = open("mesaje.decriptate", "a")
+    except IOError:
+        print("Nu pot deschide fisierul de scriere a mesajelor decriptate.")
```
În cazul în care apare o problemă ar trebui să întrerupeți execuția aplicației.

alexcoman commented on python/solutii/cosmin_crihan/caesar/mesaje.decriptate:
```diff
@@ -0,0 +1,7 @@
+ave caesar. fiecare mesaj are alta cheie.
```
Acest fișier nu ar trebui inclus.

alexcoman commented on python/solutii/cosmin_crihan/caesar/mesaje.secret:
```diff
@@ -0,0 +1,7 @@
+bwf dbftbs. gjfdbsf nftbk bsf bmub difjf.
```
Acest fișier nu ar trebui inclus.

alexcoman commented on python/solutii/cosmin_crihan/icao/from_icao.py:
```diff
@@ -0,0 +1,66 @@
+        fisier.close()
+    except IOError:
+        print("Nu s-a putut deschide fisierul cu mesajul ICAO!")
+        exit()
```
Un `return` e mai indicat.

alexcoman commented on python/solutii/cosmin_crihan/icao/from_icao.py:
```diff
@@ -0,0 +1,66 @@
+    try:
+        fisier_initial = open("icao_intrare", "w")  # deschidem fisierul pentru scrierea mesajului initial
+    except IOError:
+        print("Nu s-a putut deschide fisierul de scriere!")
```
Mai jos încercați să scrieți îna cest fișier chiar dacă a apărut o excepție.

alexcoman commented on python/solutii/cosmin_crihan/icao/icao_intrare:
```diff
@@ -0,0 +1,7 @@
+ave caesar. fiecare mesaj are alta cheie.
```
Nu ar trebui inclus acest fișier.

alexcoman commented on python/solutii/cosmin_crihan/icao/mesaj:
```diff
@@ -0,0 +1,7 @@
+ave caesar. fiecare mesaj are alta cheie.
```
Acest fișier nu ar trebui inclus.

alexcoman commented on python/solutii/cosmin_crihan/icao/mesaj.icao:
```diff
@@ -0,0 +1,120 @@
+alfa, victor, echo, 
```
Acest fișier nu ar trebui inclus.

alexcoman commented on python/solutii/cosmin_crihan/icao/to_icao.py:
```diff
@@ -0,0 +1,71 @@
+        fisier.close()
+    except IOError:
+        print("Nu s-a putut deschide fisierul cu mesajul!")
+        exit()
```
Un `return` ar fi suficient.

alexcoman commented on python/solutii/cosmin_crihan/paint/cursor.py:
```diff
@@ -0,0 +1,61 @@
+    pozitie_finala = [0, 0]
+
+    for linie in comenzi:
+        comm_val = linie.split(" ")
```
Ar trebui să verificați dacă datele valide sunt valide.

alexcoman commented on python/solutii/cosmin_crihan/paint/fill.py:
```diff
@@ -0,0 +1,77 @@
+def afisare_imagine(imaginea):
+    for linie in imaginea:
+        for element in linie:
+            print(element),
```
Ar trebui `print(element, sep=' ')`.

alexcoman commented on python/solutii/cosmin_crihan/paint/istoric.tuxy:
```diff
@@ -0,0 +1,4 @@
+SUS 5
```
Acest fișier nu ar trebui inclus.

alexcoman commented:
Fișierele din `solutii/cosmin_crihan/sync_bidirectional` nu sunt necesare.
De asemenea puteți să țineți cont și de [modificările recomandate](https://travis-ci.org/alexandrucoman/labs/builds/133110651) de pylint și pep8.

alexcoman commented on python/solutii/cosmin_crihan/unic/unic.py:
```diff
@@ -0,0 +1,67 @@
+        1 1 1 2 2 - 1
+    """
+
+    vfreq = []  # vector de frecventa
```
Ar fi mai indicat să folosiți un dicționar pentru frecvențe.

alexcoman commented on python/solutii/cosmin_crihan/tree/tree.py:
```diff
@@ -0,0 +1,31 @@
+            tree(fisier_urmator, nivel + 1)  # atunci continuam parcurgerea in adancime, pe nivelul urmator
+
+if __name__ == '__main__':
+    tree(".", 0)
```
Aveți nevoie de o linie goală la sfârșitul fișierului.

Commit:
Modificari pentru rezolvarea greselilor din aplicatii:
-newline pe ultima linie
-concatenare string-uri cu .join()
-incheiere executie aplicatie acolo unde este aruncata o exceptie
-print fara newline folosind print() din Python 3, cu end=''
-dictionar pentru frecvente (pt "unic"), in loc de lista
-eliminat linii nenecesare

Commit:
Fisierele nenecesare in commit-uri au fost eliminate.

Commit:
Fisierele sursa au fost modificate pentru a corespunde standardului PIP8.

Commit:
Fisierele sursa au fost modificate pentru a corespunde standardului PIP8.
Am sters 2 fisiere nenecesare.

Commit:
Merge branch 'cloudbase-labs' of https://github.com/astronos2007/labs into cloudbase-labs

Conflicts:
	python/solutii/cosmin_crihan/caesar/caesar.py

alexcoman commented on python/prezentare/Introducere.md:
```diff
@@ -1,982 +0,0 @@
-#Limbajul Python
```
Acest fișier nu trebuie șters din depozitul de cod.

alexcoman commented on python/solutii/cosmin_crihan/caesar/caesar.py:
```diff
@@ -0,0 +1,106 @@
+            print(litera, end='')
+
+        # intrebam utilizatorul daca mesajul e lizibil
+        plauzibil = raw_input('\n\nPlauzibil? (da/nu) ')
```
Pentru setul curent de date ai putea să găsești o soluție mai elegantă de a valida dacă mesajul a fost decriptat cu succes.

alexcoman commented on python/solutii/micu_matei/sync/functii_auxiliare.py:
```diff
@@ -1,93 +0,0 @@
-#!/usr/bin/env python
```
Nu ar trebui să ștergi soluțiile colegilor.

Commit:
Scuze pentru stergerea celor 2 fisiere :-s

Commit:
Am modificat modul de alegere al mesajului decriptat prin brute-force, la problema "caesar".

alexcoman commented:
Soluția ta nu ar trebui să aducă modificări în `python/solutii/micu_matei/sync/functii_auxiliare.py` și `python/prezentare/Introducere.md`. De asemenea ar fi drăguț dacă commit-urile care rezolvă probleme din commit-urile anterioare ar dispărea și modificările acelea ar fi aduse direct commit-urilor în cauză.

alexcoman commented on python/prezentare/Introducere.md:
```diff
@@ -930,7 +930,7 @@ open(name[, mode[, buffering]])
 ---
```
Aceaste modificări nu ar trebui incluse în acest pull request.

alexcoman commented on python/solutii/cosmin_crihan/icao/from_icao.py:
```diff
@@ -0,0 +1,72 @@
+
+Mai jos găsiți un dicționar ce conține o versiune a alfabetului ICAO:
+"""
+# pylint: disable=unused-argument
```
Nu mai este nevoie de această linie.

alexcoman commented on python/solutii/cosmin_crihan/icao/from_icao.py:
```diff
@@ -0,0 +1,72 @@
+        msg = fisier.read().splitlines()
+        fisier.close()
+    except IOError:
+        print "Nu s-a putut deschide fisierul cu mesajul ICAO!"
```
Ar fi recomandat să folosești funcția `print`.

alexcoman commented on python/solutii/cosmin_crihan/icao/to_icao.py:
```diff
@@ -0,0 +1,82 @@
+        msg = fisier.read()  # punem mesajul din fisier intr-o variabila
+        fisier.close()
+    except IOError:
+        print "Nu s-a putut deschide fisierul cu mesajul!"
```
Ar fi recomandat să folosești funcția `print`.

alexcoman commented on python/solutii/micu_matei/sync/functii_auxiliare.py:
```diff
@@ -40,8 +40,8 @@ def get_last_edit(path):
 
```
Nu ar trebui să modifici soluțiile propuse de colegii tăi.

alexcoman commented:
Commit-urile din acest pull request nu sunt legate de contul tău de Github.
